src/Control/Lens/Each.hs

```diff
@@ -221,3 +223,37 @@ instance (Ix i, i ~ j) => Each (Array i a) (Array j b) a b where
 instance (Ix i, IArray UArray a, IArray UArray b, i ~ j) => Each (UArray i a) (UArray j b) a b where
   each f arr = array (bounds arr) <$> traverse (\(i,a) -> (,) i <$> f a) (IArray.assocs arr)
   {-# INLINE each #-}
+
+-------------------------------------------------------------------------------
+-- strict
+-------------------------------------------------------------------------------
+
+-- | @since 4.20
+instance (a ~ a', b ~ b') => Each (S.Either a a') (S.Either b b') a b where
+    each f (S.Left x)  = S.Left <$> f x
+    each f (S.Right x) = S.Right <$> f x
+    {-# INLINE each #-}
+
+-- | @since 4.20
+instance (a~a', b~b') => Each (S.Pair a a') (S.Pair b b') a b where
+    each f ~(a S.:!: b) = (S.:!:) <$> f a <*> f b
```

Comment:
The use of irrefutable pattern matching on a strict pair is somewhat peculiar. Is there a reason for this?

Reply:
With the irrefutable pattern:

```haskell
>>> set each 'x' (error "foo") :: Pair Char Char
'x' :!: 'x'
```

without

```haskell
set each 'x' (error "foo") :: Pair Char Char
*** Exception: foo
```

Reply:
Ah, right. I only considered a situation where you force one of the elements of the pair (and necessarily forcing the other element of the pair to WHNF), but I completely overlooked the situation where you force _zero_ elements!